To: vim_dev@googlegroups.com
Subject: Patch 7.3.038
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.038
Problem:    v:windowid isn't set on MS-Windows.
Solution:   Set it to the window handle. (Chris Sutcliffe)
Files:	    runtime/doc/eval.txt, src/gui_w32.c


*** ../vim-7.3.037/runtime/doc/eval.txt	2010-10-20 21:22:17.000000000 +0200
--- runtime/doc/eval.txt	2010-10-27 12:23:37.000000000 +0200
***************
*** 1660,1666 ****
  					*v:windowid* *windowid-variable*
  v:windowid	When any X11 based GUI is running or when running in a
  		terminal and Vim connects to the X server (|-X|) this will be
! 		set to the window ID.  Otherwise the value is zero.
  
  ==============================================================================
  4. Builtin Functions					*functions*
--- 1660,1670 ----
  					*v:windowid* *windowid-variable*
  v:windowid	When any X11 based GUI is running or when running in a
  		terminal and Vim connects to the X server (|-X|) this will be
! 		set to the window ID.
! 		When an MS-Windows GUI is running this will be set to the
! 		window handle.
! 		Otherwise the value is zero.
! 		Note: for windows inside Vim use |winnr()|.
  
  ==============================================================================
  4. Builtin Functions					*functions*
*** ../vim-7.3.037/src/gui_w32.c	2010-10-27 12:15:28.000000000 +0200
--- src/gui_w32.c	2010-10-27 12:21:22.000000000 +0200
***************
*** 1573,1578 ****
--- 1573,1583 ----
  # endif
  #endif
  
+ #ifdef FEAT_EVAL
+     /* set the v:windowid variable */
+     set_vim_var_nr(VV_WINDOWID, (long)s_hwnd);
+ #endif
+ 
  theend:
      /* Display any pending error messages */
      display_errors();
*** ../vim-7.3.037/src/version.c	2010-10-27 12:17:54.000000000 +0200
--- src/version.c	2010-10-27 12:24:38.000000000 +0200
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     38,
  /**/

-- 
ARTHUR:        What?
BLACK KNIGHT:  None shall pass.
ARTHUR:        I have no quarrel with you, good Sir knight, but I must cross
               this bridge.
BLACK KNIGHT:  Then you shall die.
                                  The Quest for the Holy Grail (Monty Python)

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\        download, build and distribute -- http://www.A-A-P.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
